packages/peregrine/lib/talons/MagentoRoute/helpers.js

```diff
@@ -0,0 +1,18 @@
+// 301 is permanent; 302 is temporary.
+const REDIRECT_CODES = new Set().add(301).add(302);
+export const isRedirect = code => REDIRECT_CODES.has(code);
+
+// Webpack injects `fetchRootComponent` as a global during the build.
+// Depending on the environment, it may be a CommonJS or ES module.
+const warning = () => new Error('fetchRootComponent is not defined');
+const { fetchRootComponent = warning } = window || {};
+export const getRootComponent =
+    fetchRootComponent.default || fetchRootComponent;
+
+export const RESPONSES = {
+    ERROR: routeError => ({ hasError: true, routeError }),
+    FOUND: component => component,
+    LOADING: { isLoading: true },
+    NOT_FOUND: { isNotFound: true },
+    REDIRECT: relativeUrl => ({ isRedirect: true, relativeUrl })
+};
```

Comment:
I know you're just moving this code but I really don't like this map. My type senses are tingling.

Reply:
Okay, they're inline now.

Reply:
Lol yeah I wrote it and I don't like it either. All of these things used to be in that other giant chained ternary though so at least that's less of a mess.

Strikes me as a possible use case for `useReducer` since we keep track of `hasError`, `isLoading`, `isNotFound`, `isRedirect`.

Reply:
I think it's fine, honestly. We don't need to keep the response type in state; the return object is just derived from the data we get back from the two requests. This should be solid. 👍